imgcreate/fs.py

```diff
@@ -16,31 +18,40 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
-
 from __future__ import print_function
 from __future__ import absolute_import
 from __future__ import division
+
 import os
 import os.path
 import sys
 import errno
 import stat
+import shutil
 import subprocess
 import random
 import string
 import logging
 import tempfile
 import time
-from imgcreate.util import call
 
+from imgcreate.util import *
 from imgcreate.errors import *
 
-def makedirs(dirname):
+def chroottypecheck(self, command):
+    """Check for command availability in the chroot image."""
+    with open(os.devnull, 'w') as DEVNULL:
+        return subprocess.call(['type', command], stdout=DEVNULL,
```

Comment:
This seems rickety, as it depends on whether `type` actually exists. I'd prefer a Pythonic implementation here.

There's `shutil.which()` in Python 3, and it has [a backport in Python 2](https://pypi.python.org/pypi/backports.shutil_which/3.5.1) ([packaged in Fedora](https://apps.fedoraproject.org/packages/python-backports-shutil_which)).

It'd probably be a good idea to have a wrapper for command checks in chroots and command checks out of chroots to abstract the usage of either `shutil.which()` or the backport.